### ISSUE TITLE
LazyInit der MySQL-Verbindung

### ISSUE THREAD

**Issue by FrankR85** [fiduciagad/connected-n#18]
Damit der Server nicht kaputt geht, falls die MySQL-DB noch nicht gestartet ist, wäre es sinnvoll, die Verbindung erst dann aufzubauen, wenn sie gebraucht wird.

Und falls sie auch dann noch nicht aufgebaut werden kann, sollte der Server darauf fehlertolerant reagieren.

Ungefähr so: [Link zu privatem Repo](https://github.com/FrankR85/connected-n/commit/7a854dfe09b4462319a96663cb96d649c588f43f#diff-377f03fea40918f110f5dc9608a5a210)

Hintergrund: Ich überlege gerade, welche Änderungen in dem privaten Repo auch für das öffentliche relevant sein könnten und bin dabei hierüber gestolpert.

**Comment by FrankR85** [comment]
Was **noch** besser wäre: Wenn der ```MySQLListener``` völlig entkoppelt wäre und über die externen MQTT-Events, die mittlerweile ja verschickt werden, angetriggert werden würde.

Dann gäbe es zur Startzeit des Servers keinerlei Abhängigkeit zur Datenbank.

**Comment by pfichtner** [comment]
...aber dass der Broker bereits gestartet ist! ;-) 
Also gleiches Szenario in grün. Die präferierte Lösung habe ich in meinem Fork dokumentiert, übertrage ich gleich

**Comment by pfichtner** [comment]
Generell sollten alle Adapter fehlertolerant sein: Mindestens was den Wegfall einer Verbindung angeht (mit wiederholtem Versuch zum Wiederaufbau) angeht. 
Darüber hinaus ist meist auch das Ignorieren einer Nichtverfügbarkeit während des Startups sinnvoll (oder eben gar kein Versuch des Aufbaus beim Startup und lazy Initialisierung).

**Comment by FrankR85** [comment]
Genau, dann wäre es nämlich auch kein Problem, wenn der Broker noch nicht verfügbar wäre.
